src/mui/detail/Edit.js

```diff
@@ -119,12 +110,15 @@ Edit.propTypes = {
     resource: PropTypes.string.isRequired,
     title: PropTypes.any,
     translate: PropTypes.func,
+    version: PropTypes.number.isRequired,
 };
 
 function mapStateToProps(state, props) {
+    const resourceState = state.admin[props.resource];
     return {
         id: decodeURIComponent(props.match.params.id),
         data: state.admin[props.resource].data[decodeURIComponent(props.match.params.id)],
+        version: resourceState.list.version,
```

Comment:
Your state-binding is incorrect, because the `<Edit>` component grabs the `version` from `resource.list.version`. So `version` isn't at the right place in the state.

I think the position of the `version` key in the state should not depend on the resource. After all, it's just a global key that we can change to rerender the current *page*, regardless of the current ressource.

So I propose that the `refreshList` action creator doesn't take the resource as argument, and that you move the `version` key up in the state tree.